plugins/woocommerce/includes/class-wc-mobile-messaging-handler.php

```diff
@@ -0,0 +1,160 @@
+ * Prepares formatted mobile deep link navigation link for order mails
+ *
+ * @package WooCommerce\Emails
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Mobile_Messaging_Handler
+ */
+class WC_Mobile_Messaging_Handler {
+
+	const OPEN_ORDER_INTERVAL_DAYS = 30;
+
+	/**
+	 * Prepares mobile messaging with a deep link
+	 *
+	 * @param WC_Order $order order that mobile message is created for.
+	 * @param ?int     $blog_id  of blog to make a deep link for.
+	 * @param DateTime $now      current DateTime.
+	 *
+	 * @return string|null
+	 */
+	public static function prepare_mobile_message(
+		WC_Order $order,
+		?int $blog_id,
+		DateTime $now
+	): ?string {
+		try {
+			$last_mobile_used = self::get_closer_mobile_usage_date();
+
+			$used_app_in_last_month = null !== $last_mobile_used && $last_mobile_used->diff( $now )->days <= self::OPEN_ORDER_INTERVAL_DAYS;
+			$has_jetpack            = null !== $blog_id;
+
+			if ( $used_app_in_last_month && $has_jetpack ) {
+				if ( is_store_in_person_payment_eligible() ) {
+					if ( is_order_in_person_payment_eligible( $order ) ) {
+						return self::accept_payment_message();
+					} else {
+						return self::manage_order_message( $blog_id, $order->get_id() );
+					}
+				} else {
+					return self::manage_order_message( $blog_id, $order->get_id() );
+				}
+			} else {
+				return self::no_app_message();
+			}
+		} catch ( Exception $e ) {
+			return null;
+		}
+	}
+
+	/**
+	 * Returns the closest date of last usage of any mobile app platform.
+	 *
+	 * @return DateTime|null
+	 */
+	private static function get_closer_mobile_usage_date(): ?DateTime {
+		$mobile_usage = WC_Tracker::get_woocommerce_mobile_usage();
+
+		if ( ! $mobile_usage ) {
+			return null;
+		}
+
+		$last_ios_used     = self::get_last_used_or_null( 'ios', $mobile_usage );
+		$last_android_used = self::get_last_used_or_null( 'android', $mobile_usage );
+
+		return max( $last_android_used, $last_ios_used );
+	}
+
+	/**
+	 * Returns last used date of specified mobile app platform.
+	 *
+	 * @param string $platform     mobile platform to check.
+	 * @param array  $mobile_usage mobile apps usage data.
+	 *
+	 * @return DateTime|null last used date of specified mobile app
+	 */
+	private static function get_last_used_or_null(
+		string $platform, array $mobile_usage
+	): ?DateTime {
+		try {
+			if ( array_key_exists( $platform, $mobile_usage ) ) {
+				return new DateTime( $mobile_usage[ $platform ]['last_used'] );
+			} else {
+				return null;
+			}
+		} catch ( Exception $e ) {
+			return null;
+		}
+	}
+
+	/**
+	 * Prepares message with a deep link to mobile payment.
+	 *
+	 * @return string formatted message
+	 */
+	public static function accept_payment_message(): string {
+		return sprintf(
+			wp_kses_data(
+			/* translators: %s: Email link */
+				__(
+					'<a href="%s">Collect payments easily</a> from your customers anywhere with our mobile app.',
+					'woocommerce'
+				)
+			),
+			'https://woocommerce.com/mobile/payments/'
```

Comment:
Something we could consider:

```suggestion
			esc_html(
				/* translators: 1: opening link tag 2: closing link tag. */
				__(
					'%1$sCollect payments easily%2$s from your customers anywhere with our mobile app.',
					'woocommerce'
				)
			),
			'<a href="https://woocommerce.com/mobile/payments/">',
			'</a>'
```

- `esc_html()` incurs less overhead, and is also stricter
- Moves the HTML out of the translatable string (better for the translator, safer for us)
- Fixes the indentation of the translator note

Reply:
I was under the impression that this was harder for translators to deal with, but it there's already precedent elsewhere in the codebase, sounds good to me. If we go this route, it could be simplified further to:


```suggestion
			/* translators: 1: opening link tag 2: closing link tag. */
			esc_html__(
				'%1$sCollect payments easily%2$s from your customers anywhere with our mobile app.',
				'woocommerce'
			),
			'<a href="https://woocommerce.com/mobile/payments/">',
			'</a>'
```

Reply:
There could well be differing opinions on the matter, though it's definitely not without precedent (one of many cases that can be found in GlotPress):

<img width="551" alt="translating-links" src="https://user-images.githubusercontent.com/3594411/190009048-8115a0c9-94a8-4f09-96f1-8e287e2a493f.png">

I'd push for this strategy because:

- We can add classes, change the target attribute, etc, without altering the translatable string (and without having to wait on translators submitting updates).
- It lets us use stricter escaping, and so can help prevent malicious translation attacks.

> If we go this route, it could be simplified further to...

:100: